lib/view/components/events/specific_events_buttons.dart

```diff
@@ -41,87 +45,68 @@ class _SpecificEventsButtons extends State<SpecificEventsButtons>{
-                          }
-                          if (!isJoined) {
-                            EventsService()
-                                .joinEvent(e);
-                            getUpdatedEvent(
-                                e.eventID);
-                            isJoined = true;
-                          }
-                        },
-                        child: Text(
-                          isCreator
-                              ? 'Edit'
-                              : isJoined
-                              ? 'Joined'
-                              : 'Join',
-                          style: TextStyle(
-                            color: Theme.of(context)
-                                .colorScheme
-                                .tertiary,
-                            fontSize: 12,
-                            fontWeight:
-                            FontWeight.bold,
-                          ),
-                        ),
-                      ),
-                    ),
-                  ),
-                  Container(
+          Expanded(
+            child: MaterialButton(
+              shape: RoundedRectangleBorder(
+                borderRadius: BorderRadius.circular(10),
+                side: isJoined
+                    ? BorderSide(color: Theme.of(context).colorScheme.primary)
+                    : BorderSide(color: Theme.of(context).colorScheme.primary),
+              ),
+              color: isCreator
+                  ? Theme.of(context).colorScheme.primary
+                  : isJoined
+                      ? Theme.of(context).colorScheme.primaryContainer
+                      : Theme.of(context).colorScheme.primary,
+              height: 36,
+              onPressed: () {
+                if (isCreator) {
+                  Navigator.push(
+                      context,
+                      MaterialPageRoute(
+                          builder: (context) => EditEvent(
+                              e: e,
+                              imageUrl: imageUrl,
+                              edited: (updated) {
+                                setState(() {
+                                  e = updated;
+                                  widget.edited(e);
+                                });
+                              })));
+                }
+                if (!isJoined) {
+                  EventsService().joinEvent(e);
+                  getUpdatedEvent(e.eventID);
+                  isJoined = true;
+                }else{
+                  EventsService().leaveEvent(e);
+                  getUpdatedEvent(e.eventID);
+                  isJoined= false;
+                }
+              },
+              child: Text(
+                isCreator
+                    ? 'Edit'
+                    : isJoined
+                        ? 'Leave'
+                        : 'Join',
+                style: const TextStyle(
+                  color: Colors.black,
```

Comment:
Can this color please change to grey or something because it is hard to see on dark mode